stylelint.config.js

```diff
@@ -0,0 +1,225 @@
+
+    'unit-no-unknown': [true, { severity: 'warning' }],
+
+    'property-no-unknown': [true, { severity: 'warning' }],
+
+    'keyframe-declaration-no-important': [true, { severity: 'warning' }],
+
+    'declaration-block-no-duplicate-properties': [true, { severity: 'warning' }],
+    'declaration-block-no-shorthand-property-overrides': [true, { severity: 'warning' }],
+
+    'block-no-empty': [true, { severity: 'warning' }],
+
+    'selector-pseudo-class-no-unknown': [true, { severity: 'warning' }],
+    'selector-pseudo-element-no-unknown': [true, { severity: 'warning' }],
+    'selector-type-no-unknown': [true, { severity: 'warning' }],
+
+    'media-feature-name-no-unknown': [true, { severity: 'warning' }],
+
+    'at-rule-no-unknown': [true, {
+      severity: 'warning',
+      ignoreAtRules: ['mixin', 'include'],
+    }],
+
+    'comment-no-empty': [true, { severity: 'warning' }],
+
+    'no-duplicate-at-import-rules': [true, { severity: 'warning' }],
+    'no-duplicate-selectors': [true, { severity: 'warning' }],
+    'no-extra-semicolons': [true, { severity: 'warning' }],
+
+    'number-max-precision': [6, { severity: 'warning' }],
+
+    'time-min-milliseconds': [100, { severity: 'warning' }],
+
+    'unit-blacklist': [['px', 'cm', 'mm', 'pt', 'in', 'pc'], {
+      severity: 'warning',
+      /**
+       * it's not working with '["px", "dpi"]' syntax as in example in the docs
+       * so we need to use this hack with regex.
+       */
+      ignoreMediaFeatureNames: {
+        'px': [/a-zA-Z0-9/],
+        'dpi': [/a-zA-Z0-9/],
+      },
+      /**
+       * allow to use 'px' and 'dpi' in variables that has prefix 'media-'
+       *
+       * we need to use "." (dot) at the beginning of the regex to prevent
+       * handle variables values because linter doesn't read '$' value in the
+       * "/^\$media-/" regex, so we try not to match the first character
+       */
+      ignoreProperties: {
+        'px': ['/^.media-/'],
+        'dpi': ['/^.media-/'],
+      },
+    }],
+
+    'shorthand-property-no-redundant-values': [true, { severity: 'warning' }],
+
+    'value-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'property-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'declaration-block-no-redundant-longhand-properties': [true, { severity: 'warning' }],
+    'declaration-no-important': [true, { severity: 'warning' }],
+
+    'declaration-block-single-line-max-declarations': [1, { severity: 'warning' }],
+
+    'selector-max-empty-lines': [0, { severity: 'warning' }],
+
+    'at-rule-no-vendor-prefix': [true, { severity: 'warning' }],
+
+    'no-unknown-animations': [true, { severity: 'warning' }],
+
+    'color-hex-case': ['lower', { severity: 'warning' }],
+    'color-hex-length': ['long', { severity: 'warning' }],
+
+    'font-family-name-quotes': ['always-unless-keyword', { severity: 'warning' }],
+
+    'font-weight-notation': ['numeric', { severity: 'warning' }],
```

Comment:
not sure about this. maybe "named-where-possible" would be a bit more handy

Reply:
Idk, but if some guy will write `bolder` or `lighter`, but linter will be updated to allow these keywords, it can be a little confusing. With only `numeric` annotation we always are sure what exactly here means. Also, it's more consistently, because of using keywords and/or numeric format will confusedly. With this rule, you always know that "here I need to write a number" without any "but".

<img width="187" alt="Screen Shot 2019-04-03 at 3 57 55 PM" src="https://user-images.githubusercontent.com/12257282/55480571-4c951f80-5629-11e9-93df-6fda139cf487.png">

Just imagine:
```css
.selector {
  font-weight: 300;

  @respond-to(small) {
    font-weight: normal;
  }
}
```
Confused, isn't it?

Reply:
OK, makes sense, numeric is enough so